prow/plugins/project/project.go

```diff
@@ -0,0 +1,270 @@
+			log.WithError(err).Errorf("Error listing all projects under repo %s", repo.Name)
+			return err
+		}
+		projects = append(projects, repoProjects)
+	}
+
+	// Get all projects for this org
+	orgProjects, err := gc.GetOrgProjects(org)
+	if err != nil {
+		log.WithError(err).Errorf("Error getting projects for org %s", org)
+		return err
+	}
+	projects = append(projects, orgProjects)
+
+	// Get all columns for each project
+	projectColumnMap map[string][]github.ProjectColumn
+	for _, project := range projects {
+		projectColumns, err := gs.GetProjectColumns(project.ID)
+		if err != nil {
+			log.WithError(err).Errorf("Error listing all columns under project %s", project.Name)
+			return err
+		}
+		projectColumnMap[project.Name] = projectColumns
+	}
+
+	projectMap := buildProjectMap(projects)
+	projectID, ok := projectMap[proposedProject]
+
+	if proposedProject == clearKeyword {
+		projectToClear := matches[2]
+		projectToClearID, ok := projectMap[projectToClear]
+		if ok {
+			existingProjectCard, err := gc.GetProjectCard(projectToClearID, e.Number)
+			if err != nil {
+				log.WithError(err).Errorf("Error geting existing project card under project ID %d for issue/PR %s", projectToClearID, e.Number)
+				return nil
+			}
+
+			if err := gc.DeleteProjectCard(existingProjectCard.ID); err != nil {
+				log.WithError(err).Errorf("Error removing project card %d from project %s for %s/%s#%d.", existingProjectCard.ID, project, org, repo, e.Number)
+			}
+			return nil
+		}
+	}
+
+	if !ok {
+		slice := make([]string, 0, len(projectMap))
+		for k := range projectMap {
+			slice = append(slice, fmt.Sprintf("`%s`", k))
+		}
+		sort.Strings(slice)
+
+		msg := fmt.Sprintf(invalidProject, strings.Join(slice, ", "))
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	columnFound := false
+	projectColumns := projectColumnMap[proposedProject]
+	for _, c := range projectColumns {
+		if c.ID == proposedColumn {
+			columnFound = true
+			break
+		}
+	}
+
+	// If proposedColumn is not found (or not provided), add to the "To do",
+	// "To triage", or "Backlog" column. If there's no "To do", "To triage",
+	// or "Backlog" column, then add issue/PR to the first column in the
+	// projectColumns array
+	if !columnFound {
+		proposedColumn = projectColumns[0].ID
+		proposedColumnName := projectColumns[0].Name
+		for _, pc := range projectColumns {
+			if pc.Name == "To do" || pc.Name == "To triage" || pc.Name == "Backlog" {
+				proposedColumn = pc.ID
+				proposedColumnName = pc.Name
+				break
+			}
+		}
+		msg := fmt.Sprintf(invalidColumn, proposedProject, proposedColumnName)
+		return gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, msg))
+	}
+
+	// Move this issue/PR to the new column if there's already a project card for
+	// this issue/PR in this project
+	existingProjectCard, err := gc.GetProjectCard(projectID, e.Number)
+	if err != nil {
```

Comment:
Will you get a not found error here?

Reply:
Yeah probably will, in which case I shouldn't check for err here because we will move onto creating a new project card